### DISCUSSION TITLE
DateTime.Now & Subscriptions

### DISCUSSION THREAD

**Discussion by tobiaszuercher** [ravendb/ravendb#15723] [Q&A] [answered]
hi
i'm aware that DateTime.Now can't be used in indexes as indexes need to be static.

i was wondering if there is a way around that with data subscriptions?

a use case i see often is e.g. to process a document after a certein date. e.g. all documents where doc.ExpiresAt < DateTime.Now of a collection.

what i end up doing is a .net hosted service that periodically queries raven. would be much nicer with datasubscriptions since datas subscriptions provide some nice things around it.

only other way i see is with the Changes API. but there i'm not able to filter.

am i missing something?

i was naive enough to simply try it out quickly:

```store
            .Subscriptions
            .CreateAsync<MyDoc>(u => u..ExpiresAt < DateTime.Now, token: stoppingToken,
                options: new SubscriptionCreationOptions()
                {
                    Name = workerName,
                });
```

docs with ExpiresAt = null actually appear then in the worker. but if i set any "real" datetime value it doesnt work.

**Comment by garayx** [comment, marked as answer]
Hi, 

It won't work the way you want, the issue is that you try to do a query on time, but in subscription we are not running the query on documents, we are matching the document to the query, and then consider the future ones, so if the document didn't match the subscription criteria, it won't evaluate again.

You probably can achieve what you want, using the refresh feature:
1. https://ravendb.net/learn/webinars/using-ravendb-as-a-queuing-infrastructure 
2. https://ayende.com/blog/189473-C/refreshing-documents-in-ravendb
3. https://ravendb.net/docs/article-page/5.4/csharp/server/extensions/refresh

**Reply by ayende** [reply]
Yes, the proper way to handle that is `@refresh`, intended specifically for this scenario
